### ISSUE TITLE
Code Review (Bartosz Wilkowski 19697964)

### ISSUE THREAD

**Issue by BartoszWilkowski88** [25713655/CMP1903M-Assessment-1#2]
# Code Review
I will be making code review for the whole code in general rather than each class individually. I recommend taking any comments I make lightly and perhaps looking into them further as I am fairly new to programming myself therefore, my comments could be wrong.

### Issues
When running the Test file option within the program, it gives the correct result however, that is because you've entered the text from the file manually beforehand and stored it within testText variable. If you instead loaded the test file with file.readAllText() the results would not be accurate. Below you can see an example of me loading a different file, displaying incorrect results.
![image](https://user-images.githubusercontent.com/102090465/160088711-78d2d0b1-8f3e-4af5-be36-b30264de3f1d.png)
This is because your program doesn't recognise asterisk as an end of entry. As well as your program counts ellipsis(...) as 3 sentences rather than 1. I also have the ellipsis issue within my program and currently am working on a fix myself therefore, cannot give you a way of fixing it yet. However, in a review that has been done on my code, another student has mentioned the use of [Regex](https://regextutorial.org) refer to the link for tutorial. 
That seems to be a good way to deal with the ellipsis issue. Below is an example of manual text input to show its the same result as file input.
![image](https://user-images.githubusercontent.com/102090465/160090158-2993e181-b84f-4d8e-a0b4-b8589f84927c.png)
From the above example, based on my understanding of the brief the program should only treat 'Hello' as the sentence to analyse 
### Long words
I think your long words file generator is not working at all, perhaps you didn't upload your final code to repo? Regardless of what the reason is, it didn't work for me so I can't comment much on that.
### Positive
Error handling seems exceptional, haven't found a way to break the code, good job.
Method and class usage is great.
### Negative
Code is not very efficient, one example is you do a if (text != "") break; multiple times within the same loop, which ultimately does the job so its fine however, if the project was 10x the size of this one you could end up with essentially 100's of lines that could of been narrowed down to 20 or less. Reason why you should try to simplify your code where you can is so that its easier to read in larger projects. Given the way that you've written your code the only thing I can think of to avoid this is by adding a method and instead of writing out the if statement every time you could just call the method instead. This way could generally be used for any repetitive code. You could also try using guard clauses.
The code I'm talking about is: 
![image](https://user-images.githubusercontent.com/102090465/160090939-dca40d42-fb50-480b-85cb-9dbcdb08a2ce.png)
in the example above since you know your input will be 1, 2 or 3 and nothing else. You could use a guard clause such as:
``` 
While(choice != "1" && choice != "2" && choice != "3")
{
               Console.WriteLine("Invalid Input")
{
```
This is just a minor option that I'm suggesting for the future, by all means I do not think it is required for your code as your code does the job its supposed to anyway.
### Commenting
Commenting overall, is good but not consistent. In analyse your commenting is great meanwhile in the other classes its either minimal or non existent.

### Conclusion
In conclusion I think the program still needs some work as Long Words didn't work for me. Also the test file is not actually loaded in but instead manually stored in a variable. The program also does not recognise asterisks as an end of entry and ellipsis is considered 3 sentences. Aside from that you've shown some exceptional coding practices as well as some inefficient ones. Overall though I believe there are more positives than negatives within your code especially due to the fact that I wasn't able to break your program due to your exceptional error handling, when I look at the exception handling code I do believe it could be written more efficiently as there is a lot of repetitiveness involved however, as far as the functionality itself goes, its perfect. 
Good job mate and good luck!
